### PR TITLE
print config and version info after I_InitPrintf()

### PR DESCRIPTION
**Pull request by rfomin**

### CODE REVIEW

File: src/d_main.c [hunk 1]
```diff
@@ -2356,10 +2363,14 @@ void D_DoomMain(void)
...
     WritePredefinedLumpWad(myargv[p+1]);
 
-  I_Printf(VB_INFO, "M_LoadDefaults: Load system defaults.");
   M_LoadDefaults();              // load before initing other systems
+
   I_InitPrintf();
 
+  PrintVersion();
+  I_Printf(VB_INFO, "M_LoadDefaults: Load system defaults.");
+  I_Printf(VB_INFO, " default file: %s", defaultfile);
```
Comment (fabiangreffrath) [inline src/d_main.c:2372]: Now, hieß dies it look if `M_LoadDefaults()` cannot find a config file?

Comment (rfomin) [reply src/d_main.c:2372]: Then a new woof.cfg file will be created. I think we always load the config.

Comment (rfomin) [reply src/d_main.c:2372]: A little awkward:
```
Warning: Cannot read c:\develop\woof\build\src\woof.cfg -- using built-in defaults
IWAD found: C:\games\OLD\DOOM\WAD\doom2.wad
DOOM II version

M_LoadDefaults: Load system defaults.
 default file: c:\develop\woof\build\src\woof.cfg
```

Comment (fabiangreffrath) [reply src/d_main.c:2372]: Hm, that doesn't make too much sense. 🤔

**Comment by fabiangreffrath** [comment]
Let's see. It's not a problem if `M_LoadDefaults()` reporting comes first before `PrintVersion()`.

`M_LoadDefaults()` has two possible outcomes. Either "config file not found" which would be a warning and would be printed anyway unless explicitly configured otherwise (but we don't have a config yet at that point). Or "config file found" which we could print *after* the config has been parsed and thus know if this line is meant to be printed or not.

**Comment by rfomin** [comment]
> `M_LoadDefaults()` has two possible outcomes. Either "config file not found" which would be a warning and would be printed anyway unless explicitly configured otherwise (but we don't have a config yet at that point). Or "config file found" which we could print _after_ the config has been parsed and thus know if this line is meant to be printed or not.

I tried to fix this in https://github.com/fabiangreffrath/woof/pull/1194/commits/a47a3fc800e24938120ea81553e87776382a764a. Maybe it's overcomplicated.

**Pull request closed without merging**